### PR TITLE
Add command to debug context filters

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes CODY-1341

Add a command to overwrite context filters. I've for now gated it to only show up when verbose debugging is enabled however that doesn't seem to be working well 🤔 

## Test plan

```
{ "include": [], "exclude": [{ "repoNamePattern": "^github\\.com/sourcegraph/sourcegraph" }] }
```

https://github.com/sourcegraph/cody/assets/458591/4fda5932-a708-457f-9a55-072cd42d9b08

### CODE REVIEW

File: vscode/package.json [hunk 1]
```diff
@@ -565,6 +565,11 @@
         "icon": "$(search)",
         "title": "Natural Language Search Code (Beta)",
         "when": "cody.activated && workspaceFolderCount > 0"
+      },
+      {
+        "command": "cody.test.set-context-filters",
+        "title": "[Internal] Set Context Filters Overwrite",
+        "when": "!cody.devOrTest"
       }
```
Comment (philipp-spiess) [inline vscode/package.json:573]: Anyone know why this would show up no matter if I invert this check or not? 🤔

Comment (umpox) [reply vscode/package.json]: Did you fix it? Curious why this was happening

Comment (umpox) [reply vscode/package.json]: Wondered if you needed to manually set it in the `commandPalette` section of package.json too

**Review by umpox — APPROVED** [review]

File: vscode/package.json (hunk 1 above)
Comment (philipp-spiess) [reply vscode/package.json]: Still looking haha let me try that

Comment (philipp-spiess) [reply vscode/package.json]: Ah yes that's it nice! TY.
I find this json config SO SO confusing 🙃

Comment (philipp-spiess) [reply vscode/package.json]: I think `when` here doesn't do something actually? There isn't even a JSON description? Why does every action define one though 🤔

Comment (umpox) [reply vscode/package.json]: I vaguely remember needing to declare the command in one place, and then exclude it from the command palette in another place. I thought this is the same issue, I think you need both?

Agree it's so confusing

Comment (philipp-spiess) [reply vscode/package.json]: Docs are not very clear they mention you have to use `when` but not where `when` is defined: https://code.visualstudio.com/api/references/contribution-points#contributes.commands

If you hover over `when` in `commands`, you get no title (suggesting it's not part of the spec)

What a mess

**Pull request merged**